### DISCUSSION TITLE
Lc Value vs Simple Ratio •• AND •• Bridge-PCA

### DISCUSSION THREAD

**Discussion by Myndex** [Myndex/SAPC-APCA#42] [DOCUMENTATION]
Hi Bruce! @bruce-usab 

> _Relatedly, and maybe this should be its own discussion, I would like to understand better the statement that Lc 60 is sort of like 4.5:1._

Let me explain briefly _(well as briefly as I... um... well... okay.... let me... explain)_

## _Bridge-PCA_
First, let me introduce you to **Bridge-PCA** — this is a 100% backwards compatible replacement of the WCAG_2 contrast math. By 100% I mean that *if you pass Bridge-PCA you automagically pass WCAG_2* (using the same fonts as WCAG_2). **To emphasize:**
### if you pass Bridge-PCA you _automatically_ pass WCAG_2
Weeee! But there is no free lunch. To be backwards compatible, the contrasts have to be forced higher than technically needed when using the full APCA and font lookup table. The trade offs are "easy to use" vs "most flexible for design". I can make the conformance as easy to use as WCAG2, or more functionally flexible, as what I've shown for APCA.

I explain this more completely below, now to answer your main question:

---------
## The _"exceeds"_ vs _"sort of like"_ thing
Sooooo.... Trying to keep this brief. How does Lc 60 relate to 4.5:1 or 3:1...

The "perceptual middle" grey has traditionally been considered 18% (18% grey card), but like all of this, the reality is more complicated. In sRGB, 18% is about #777 ... and this is also what _WCAG_2 math_ considers middle contrast.

And while it is easy to assume that "middle grey" would also be middle **_contrast_**, well....
### ...nope.
Middle _**contrast**_ shifts around based on a pile of the usual psychophysical factors.

On a typical monitor, and depending on spatial frequency, the perceptual middle tends to be around `#999` to `#AAA`, as dark as ~`#888` and as light as ~`#bbb` depending on stimulus size, ambient, etc. **`  #a0a0a0 `** is a good "rule of thumb" middle between black and white text, if you want a place to flip from white text to black text based on the BG. For proof of this I made a CodePen:
### Great, so what about APCA vs WCAG2?
As we know, WCAG_2 is centered on old technology and uses a simple linear ratio that does not follow human perception. For any given contrast value, for the **_highest actual perceived contrast,_** the lightest color **must be white #fff**. As the lightest color gets darker, the 1.4.3 contrast math is increasingly inaccurate relative to how we see.

APCA is perceptually uniform for a given contrast value across the available range, with a very slight boost in very dark colors, there to assist with misadjusted monitors, flare, and other issues.

### There are two ways to compare WCAG\_2 Ratio vs APCA Lc:

- What Lc value meets or exceeds the max WCAG\_2 contrast for a given ratio, OR
- What Lc value is functionally equivalent to a useable WCAG\_2 ratio.

- _"Exceeds WCAG_2"_ This is "Bridge-PCA"
    - Backwards compatible (pass APCA, autopass WCAG2).
    - Is more "brute force" in terms of forcing higher contrasts than might be needed.
    - Allows for simpler conformance.

- _"Functionally Equivalent"_ This is "APCA for W3"
    - Is more design flexible, but _not_ backwards compatible.
    - Design flexibility comes from recognizing use cases and spatial frequency.
         - Use case or spatial frequency may indicate a higher or lower contrast than demanded by WCAG_2
         - Readability is improved by putting higher contrast color pairs where they are needed.
         - Flexibility is improved by not demanding excess contrast where it is not needed.
         - Perceptual middle contrast and polarity is correctly assessed.
     - More complex conformance models and testing.
         - Accurate font and weight settings are more important.

- **Lc 60 Exceeds WCAG 3:1**
- **Lc 75 Exceeds WCAG 4.5:1**
- **Lc 90 Exceeds WCAG 7:1**

The "exceeds" set exceeds WCAG_2 over the entire range. So much so the AA set is able to relax slightly for darker colors, but still superior to 1.4.3. The AAA set is superior to 1.4.6

- **Lc 45 Functionally Equivalent to WCAG 3:1**
- **Lc 60 Functionally Equivalent to WCAG 4.5:1**
- **Lc 75 Functionally Equivalent to WCAG 7:1**
- **Lc 90 Functionally Equivalent to WCAG 10:1**

The Functionally Equivalent values are based on a narrow range with a center color ~ #9e9e9e (per the G-4g scaling) or luminance ~34Y to ~42Y.  Outside of this very narrow range only APCA remains perceptually accurate, the old method does not. The following table demonstrates the "Functionally Equivalent" relationships.

<img width="640" alt="Screen Shot 2021-12-15 at 11 03 35 AM" src="https://user-images.githubusercontent.com/42009457/146668480-7b20380c-74c0-460f-a802-f7b17a02047a.png">

### _"Engineered on Purpose"_ 😁
The SAPC math was engineered to put the key values at the easy to remember levels of **Lc 15 30 45 60 75 90** etc. 

I specifically wanted Lc 45, Lc 60, Lc 75 to line up at some point with WCAG 2.x's 3:1, 4.5:1, and 7:1  to be able to say that "Lc 60 is _about_ like the old 4.5:1" to help in moving toward this new guideline.

Because APCA is perceptually uniform within reasonable tolerances, a change by a certain Lc amount has a similar perceptual change regardless of the present value (except of course when one of the colors is clipped at #000 or #fff). In the current scaling, increasing contrast by adding Lc 15 is _approximately_ equivalent to the contrast change of a 400 weight font to a 700 weight font, when the font is smaller than about 32px. 

The G Series exponents demonstrate that the old WCAG2 spec is completely out-of-line with perception. Darker color pairs that WCAG2 incorrectly passes will fail under APCA, and light text on dark backgrounds that WCAG incorrectly fails will correctly pass under APCA. Ultimately, APCA will provide 50%+ more colors, and do so correctly per use cases and spatial frequency.

-----
CONFORMANCE INFO ON THE BRIDGE-PCA VERSION
-----
## Bridge to the Advanced Perceptual Contrast Algorithm
Current Version: **0.1.0 4g** (w3) _betafish_

Bridge PCA is a contrast assessment method based on APCA & SAPC technology for predicting the perceived contrast between sRGB colors on a computer monitor. It has been developed as an assessment method to facilitate better readability while maintaining backwards compatibility with WCAG_2 contrast (1.4.3) accessibility standards relating to content for computer displays and mobile devices. The intention is to improve readability and understandability of content.

## WCAG_2 Bridge-PCA Font Use

**Large Font:** 24px normal weight or 18.7px bold, or larger.
**Thin Font:** 300 weight.

-----
### AA
***SHOULD***
- **Lc 15 (W 1.3:1)** Minimum for disabled elements (not hidden).
- **Lc 30 (W 1.8:1)** Minimum for incidental text such as placeholders.
- **Lc 45 (W 2:1)** Minimum for logotypes.

***SHALL***
- **Lc 60 (W 3:1)** Large font only, no body text. Non-text okay.
- **Lc 75 (W 4.5:1)** 16px minimum for body text, 12px minimum otherwise
- **Lc 90 (W 7:1)** 14px minimum body text, 10px minimum otherwise

***MAY***
- If the lightest color is *darker* than the equivalent of #d8d8d8, the minimum Lc value can be reduced by Lc 10, but lowered to no lower Lc 45. Does not apply to thin fonts.

-----
### AAA
***SHOULD***
- **Lc 30 (W 1.8:1)** Minimum for disabled elements (not hidden).
- **Lc 45 (W 2:1)** Minimum for incidental text such as placeholders.

***SHALL***
- **Lc 60 (W 3:1)** Minimum for logotypes and essential non-text.
- **Lc 75 (W 4.5:1)** Large font only, no body text.
- **Lc 90 (W 7:1)** 16px minimum for body text, 12px minimum otherwise

***MAY***
- **Lc 90 (W 7:1)** Suggested maximum for very large and bold elements.

-----
Unlike the main APCA, BridgePCA is all about "emulating" WCAG\_2 contrast. So, BridgePCA is a like-for-like replacement of the quirky WCAG\_2 contrast math.

Like APCA, BridgePCA reports results as Lc (Lightness Contrast), and it's super easy to convert to WCAG\_2 ratios:

- **Lc 60 exceeds WCAG 3:1**
- **Lc 75 exceeds WCAG 4.5:1**
- **Lc 90 exceeds WCAG 7:1**

***DIFFERENCES:*** for best results, send the text color to the text input of the tool. Bridge PCA is "polarity sensitive," even though WCAG\_2 is not. For light text on a dark background, the result will have an R for "reverse" after the LC value. N means "normal".

**No Free Lunch:** while BridgePCA corrects the many false passes and improves readability, the cost is that there is reduced design flexibility due to the fact that to maintain backwards compatibility, some contrasts are forced higher than they actually need be.

BridgePCA has some minor internal adjustments to align with some of the more incorrect aspects of WCAG\_2 contrast math. In order to be backwards compatible, BridgePCA will not forgive the false-fails of WCAG\_2, but BridgePCA **will** correct the many false passes which vastly improves readability.

But if you need a standards compliant method that also improves readability this is it. If on the other hand you do not need to abide by the letter of any particular standard, you may want to consider the more flexible full APCA solution.

Thankyou

Andy

**Comment by bruce-usab** [comment]
Thanks @Myndex , I will come back to this more, but four notes-to-self kinds of things.  These are all just my personal opinions, and and I am not speaking for USAB or AG WG.

(1) Keep in mind that a should/shall/may approach is actually counter-productive in the context of crafting 2.x SC phrasing.  It is good for a11y _guidance_ and for discussing things generally, and fits WCAG3.

(2) That said, as a plug-and-play replacement for 1.4.3, this seems very promising to me:

> &bull; L<sup>c</sup> 60 (W 3:1) Large font only, no body text. Non-text okay.
> &bull; L<sup>c</sup> 75 (W 4.5:1) 16px minimum for body text, 12px minimum otherwise
> &bull; L<sup>c</sup> 90 (W 7:1) 14px minimum body text, 10px minimum otherwise

(3) Does L<sup>c</sup> 75 permits both white and black on a certain hues?  (I think so.) Could you please write up something more about that?  White and black on orange is a good example.  Is there a blue?  How about a green?  From lack of experience, I don't have an intuitive feeling for the range of hues where L<sup>c</sup>75 is possible for both white and black text.

Your have provided many of these kinds of samples, but I would like to have a page or discussion thread just for this three-color challenge.  My understanding is that  L<sup>c</sup> 75 white and black on any middle gray is not possible, so please clarify that.  Not having a middle _gray_ that supports L<sup>c</sup> 75 (for both black and white text) is probably not a blocker to adoption.

Historically, that 2.x 4.5:1 allowed for black (and white) on a middle gray was important to its early adoption and acceptance.  But the contrast (for black on those middle grays) is quite apparently not great, so it is very good to have a better algorithm.  WebAIM link example: [white and black on gray](https://webaim.org/resources/linkcontrastchecker/?fcolor=000000&bcolor=757575&lcolor=FFFFFF)

2.x 4.5:1 allows a very limited range where this works.  Some hues are possible, but really they are fairly characterized as all being a middle gray.

(4) I perceive a need to figure out how to get some real-world testing by end-users with [Retinitis Pigmentosa](https://www.nei.nih.gov/learn-about-eye-health/eye-conditions-and-diseases/retinitis-pigmentosa) and L<sup>c</sup>75 reds and oranges.  I have some ideas of where to start asking, and I plan to make an inquiries in 2022, but before that, I would like to have more confidence that L<sup>c</sup> 75 is the value which should be subject to particular scrutiny.

**Reply by Myndex** [reply]
Hi Bruce! @bruce-usab   Thank you these are excellent questions, some have led to recent refinements last couple days, others may have been answered elsewhere and I'll link... This has lead to my realization that over the last nearly three years my work is now so scattered about that it's even less coherent than ever. UGH!!! Collating and bringing everything together is a priority for 2022.

In order to make my posts more readable, I'm going to try to just link to the larger explanations.

> _(1) Keep in mind that a should/shall/may approach is actually counter-productive in the context of crafting 2.x SC phrasing. It is good for a11y _guidance_ and for discussing things generally, and fits WCAG3._
> _(2) That said, as a plug-and-play replacement for 1.4.3, this seems very promising to me:_
> > _• Lc 60 (W 3:1) Large font only, no body text........_

Got it. I revised into the "Drop In" and then two "enhanced" levels. The drop in is literally just "replace the math, and LcX = Y:1" [as explained here.](https://github.com/Myndex/bridge-pca#wcag_2-drop-n-go).

Right after that,  [I listed "Bridge-PCA Enhanced"](https://github.com/Myndex/bridge-pca#wcag_2-bridge-pca-enhanced-scs) which is an AA enhanced and an AAA enhanced, which adds the "a11y guidance".

> _(3) Does Lc 75 permits both white and black on a certain hues?_

Yes and no. For instance, fully saturated blue #00f should not be used with black, nor should full red #f00 and will fail basic APCA. Full red #f00 does not fail WCAG_2, one of its downfalls. In fact WCAG_2 rates #f00 against black as 5.25:1. APCA though puts it at 37 below the cutoff for all fluent text (45). 

There is literally not enough available contrast in a typical sRGB display to have two high-contrasts of body text sized stimuli on top of each other with a common middle. It might be possible in HDR, but with high expected fatigue.

On the https://www.myndex.com/SAPC/  site, there is "research mode" — click research mode and select the middle contrast experiment that demonstrates this.

### ._...Andy Tangent Alert...._
This by the way is an important proof to how WCAG_2 does the opposite of what it is claimed to do, and in this case is harmful to protanopia. While it is true that in sRGB a protan will see the luminance of the red primary, it is 35% or more darker. This makes #f00 text on black unreadable for protanopia unless it is very large (60px ish) But WCAG_2 says 5.25:1, and for protanopia, red #f00 is fine against white, but WCAG_2 fails it at 3.99:1

WCAG_2 is backwards, literally. Here's a comparison chart from some very early studies in 2019, comparing different maths. This is sort of equivalent to the current Lc 60 — WCAG2 just flys off in the wrong direction as the colors get darker.

<img width="640" alt="Contrast Methods Compared Chart 2019-11-01" src="https://user-images.githubusercontent.com/42009457/147395624-b58b8ab7-e0b2-402b-9a4f-d8c395ca7bd5.png">

Compared to other common contrast maths WCAG_2 performs worst (literally flying off scale), and when comparing to the needs of CVD, WCAG_2 performs the opposite of what it needs to. I have [visual examples in this article from a year ago.](https://gist.github.com/Myndex/c30dba273aa5eca426ad9f5200917c9d#the-lighter-side-of-dark-backgrounds)
### _END TANGENT ALERT_

## Flip For Color!
> _You have provided many of these kinds of samples, but I would like to have a page or discussion thread just for this three-color challenge. My understanding is that Lc 75 white and black on any middle gray is not possible, so please clarify that. Not having a middle _gray_ that supports Lc 75 (for both black and white text) is probably not a blocker to adoption._

I discuss this "middle point" more in my article [**Let's Flip For Color**](https://gist.github.com/Myndex/e1025706436736166561d339fd667493#lets-flip-for-color)  But specifc to adoption and "three way" as a design issue:

There is no reason it should be a blocker for adoption, because that has never been a "thing" in classical design.

For body text, there is no need for the "three way"; that has never been a "design need". The "multi-way" color is more about text vs a background **versus** object vs background, i.e. a large button. A large button shape (ignoring text) needs much less luminance contrast than text, which is thin. This is the key importance of spatial frequency.

That APCA is designed to consider spatial frequency is where the design flexibility comes in, and it is relevant to classical design rules and relevant to perception. The need for this to this to this to... etc is not (and functionally can not) be about a pair set of three colors.

- Contrast perception is first about spatial frequency (weight and size of font), then about luminance, then about a closet full of other things. 

- Visual acuity is also first about spatial frequency (size, weight), then about luminance.

**So for three-way:** is there ever any case wherein body text is going to be on top of other body text, that needs a third color? No, but there is a common case of background A, and sub-background B (i.e. button) and text on the button. BUT still, where is the three way? there is the need for contrast between the background and the button shape, which is low because the button is a large, thick object, and then there is the need for the text against the button. So, even here, there is no functional need for "three way". The button can easily be Lc 45, and often as low as Lc 30. The text needs to be at least Lc 60, or Lc75, and maybe Lc90. The APCA range is 105+

But the "contrast center" where there is an equal amount of contrast above and below a common "mid grey" is about Lc 54-ish.
(Note: I'm working on a very minor adjustment here regarding contrast centers). 

**I can't change the physics of light, nor how we perceive it — I can only model the _reality_ of light and perception.**

> _Historically, that 2.x 4.5:1 allowed for black (and white) on a middle gray was important to its early adoption and acceptance. But the contrast (for black on those middle grays) is quite apparently not great...._

Yes, unfortunately on the dark half it's unreadable...!

While there is the "classical" middle grey of 18Y (18% grey card) that is not necessarily middle contrast, though that does happen to be where WCAG_2 puts the middle around #777. The proof and the reasons why are the subject of a paper I am working on. And it's a deeper rabbit hole that I initially expected, but I do discuss more in the Flip for Color article I linked above.

In addition to the above article on Flip for Color I have a live [CodePen that can be played with ](https://codepen.io/myndex/full/RwZJyPR)that demonstrates this. You'll see in the mid grey area the text gets larger to compensate.

**SIDE NOTE:** Rather than focus on publishing papers, I've focused on presenting prima facia evidence and live examples that people can play with and see for themselves. Based on some recent controversies, I'm reducing some of these key discoveries to academic papers to address this. 

> 2.x 4.5:1 allows a very limited range where this works. Some hues are possible, but really they are fairly characterized as all being a middle gray.

Well, again, this is a downfall — it's not the "3 way" that is as important as readability, and the recognition that some objects like a button shape need much less contrast than text. Once we realize that we need to establish the needs based on spatial frequency _FIRST_, then use case, then color, hue, luminance, all falls into place in a way that is very useable and very readable.


> (4) I perceive a need to figure out how to get some real-world testing by end-users with Retinitis Pigmentosa....I have some ideas of where to start asking,.... I would like to have more confidence that Lc 75 is the value which should be subject to particular scrutiny.

I really want to gather some data here too, and the published literature on RP is weak. I definitely want RP individuals involved in the 2022 studies.

### As far as should Lc 75 be the focus
of the five primary levels it's probably going to become the catch all. The five big-uns are 30 45 60 75 90, with 15 and 105 being the "ends" and this is for a typical sRGB display in typical indoor ambient conditions similar to the SAPC standard observer.

### Not Set In Stone:
- 15 - invisibility level, minimum for discernible (form or button outlines, _large_ focus indicators)
- 30 - minimum for "understandable" large, non text, not readable (placeholder text that is not a label, disabled controls)
- 45 - minimum for large text, minimum for smaller spot readable text
- 60 - minimum for fluent text except very large bold text
- 75 - minimum for body text, preferred for fluent text
- 90 - preferred for body text, suggested maximum for very large AND bold elements.
- 105-approximate top scale for standard dynamic range display.


### What is Lc 75 ?
The driving force on much of this, as indicated in a number of posts I've made in the LVTF and on the VC Wiki, is driven by the research of Bailey/Lovie-Kitchin, and also Legge, in readability. The current settings are such that every time I think it needs to be adjusted one direction or another, moving to another environment indicates adjusting the other way. In other words it represents "close to ideal compromise" while keeping the math and overall guideline simple.

And still some minor voices are calling it too complicated. But in a semi-automated design system, or testing system, it is easy to implement.

As such, there are multiple "levels" emerging:

- Bridge-PCA: Straight WCAG_2 drop in, with optional enhancements, and is open source.

- APCA-W3: Just the basic contrast value, with conformance and font specifications pending more work for WCAG_3. This is what we've been using and showing, with a basic lookup table (to be replaced with a parametric curve set for software). 
    - I am working to make a "simple base" where a lookup table is not specifically needed.
    - and a separate one with a more complete lookup for added flexibility in automated design systems.
    - APCA-W3 is open source.

- SAPC, SACAM, APCA<sub>specific versions</sub> are not part of what is going into WCAG_3, but relate to other specific applications and uses, as it became clear that the need for better contrast metrics is much wider. Some of this work may finally help in some needed areas of the ADA, like, uh, signage... which is in sad sad shape.....


All for now, 

Thank you!

Andy

**Reply by bruce-usab** [reply]
> Is there ever any case wherein body text is going to be on top of other body text, that needs a third color?

I cannot think of any body-text-on-body-text example.  But there are plenty of examples where three colors of good mutual contrast are quite important.  Any map or graph really, but in general, than requires four colors with good mutual contrast. which is not mathematically possible.  One has to settle for 3:1 and use [other approaches for maps](https://colorbrewer2.org/).  That said, having black, white, and a third color opens up lots of design choices.  I think a Word Cloud is a good example where 4.5:1 should be the minimum expectation.  Not body text, but text that is meant to be read easily.

> For body text, there is no need for the "three way"; that has never been a "design need". 

I respectfully disagree with this assertion.  A classic example is traditional old-school BBS style web page.  In the 90s this would have meant a noticeable background color.  If I have a page giving homage to tigers or Halloween, I want an an Orange field.  Black text with linked text and 1em headers in White.

> The "multi-way" color is more about text vs a background versus object vs background, i.e. a large button.

I agree that multi-way is <q>more about</q> but it is not _only_.

**Reply by Myndex** [reply]
Hi Bruce @bruce-usab 

I think we are saying almost the same thing. I am not saying that multi-level color is not needed, what I am saying is that body text is (traditionally, if you want people to actually read it) is handled uniquely relative to any other element type.

If "multi way" in involved with body text, the body text is traditionally given a cutout-area or some kind of hold-out so that high contrast can be maintained. 


> .._.there are plenty of examples where three colors of good mutual contrast are quite important. Any map or graph really, but in general, than requires four colors with good mutual contrast. which is not mathematically possible. One has to settle for 3:1 and use [other approaches for maps](https://colorbrewer2.org/)._ 

Yes, but map areas are not small text. Map areas are by definition, areas. Map _areas_ are not "high spatial frequency", and are in fact large to the post of being considered low spatial frequency (except hard edges) where chroma/hue contrast is strong, and can be stronger than luminance contrast (CVD not withstanding of course). 

TEXT especially body text is small and thin and at a very high spatial frequency. Map areas or other areas of a color or lightness is a low spatial frequency, and the contrast sensitivity needs are completely different.

I know you've seen this contrast sensitivity chart before, but for review or anyone else reading along at home:

### _click for full size_
<img width="400" alt="contrast sensitivity chart showing font sizes" src="https://user-images.githubusercontent.com/42009457/148161314-69a10166-0ad8-4fc5-a7d1-17101c373218.png">

----

What that chart shows is that the needs of body text vs large bold text, or areas of color can be as much as **two orders of magnitude.** Now, at higher contrasts that curve flattens a but due to contrast constancy. But the point is the contrast needs to text and that for areas of color like on a map are essentially unrelated. ANd then this leads into the discussion of if you need to "discern" differences, or "understand" differences.

- By discern, this implies the basic "can clearly see there is a difference or division".
    - This means the edge of a form field, or a divider, disabled control, etc.
    - The range of Lc15 to Lc30 is based on this, the "minimum discernment level" taking into account impairments, and with a lot of restrictions of course.
    - Standard vision can see well below Lc15, but a profound vision impairment limits the threshold to about 10%, which correlates about to Lc10, Lc15 is thus a reasonable "invisibility level". 

- By understand, this implies "can clearly see a relationship or information however coded"
    - This means an icon, information coding as in two colors where the colors carry a meaning such as "farm areas are green, city areas are purple", or to easily see the difference (focus/non focus).
    - This starts at Lc 30 and up.
    - FWIW, The 3:1 is not a thing here for areas of color or large non text stimuli, and it has no empirical support — in 1.4.11, the only support listed are a couple of informal emails, and there is conflating bob-text with the needs for text contrast for readability, which is entirely different. Unfortunately it is this kind of thing that can impact credibility of the guideline, and/or cause pushback.

> _That said, having black, white, and a third color opens up lots of design choices. I think a Word Cloud is a good example where 4.5:1 should be the minimum expectation. Not body text, but text that is meant to be read easily._

Yes and it can if the font is large enough — again with the chart above, once things are smaller than 20 cycles per degree or so, contrast needs skyrocket. And the skyrocketing happens a bit sooner for impaired visions.


> > For body text, there is no need for the "three way"; that has never been a "design need".
> I respectfully disagree with this assertion. A classic example is traditional old-school BBS style web page. In the 90s this would have meant a noticeable background color. If I have a page giving homage to tigers or Halloween, I want an an Orange field. Black text with linked text and 1em headers in White.

Okay, but a classic BBS is a poor example of good design.... 🤣 I remember those days lol _(300 baud acoustical coupler uphill both ways!!)_

I can't change the laws of physics of light, and I can't change how our perception evolved. I can only report what the limitations are.

Screen shot of https://codepen.io/myndex/full/RwZJyPR near middle contrast the font is enlarged...

<img width="1439" alt="FancyFontFlippingScreenshot" src="https://user-images.githubusercontent.com/42009457/148189630-61e407ef-db9f-4a6d-a0f7-b30e8cedcbdc.png">


> > The "multi-way" color is more about text vs a background versus object vs background, i.e. a large button.
> I agree that multi-way is more about but it is not _only_.

Yes I agree, more about... but if you have a column of body text, it likely will need a hold-out area on a complex page, OR use a large enough font...

### map tool...
BTW that's an interesting map tool, in playing with it I found 5 and 6 colors that worked and also allowed borders and roads, and were okay for CVD... but the tool does not allow individual adjustment of colors, which I found to be necessary when considering the overlays of single color roads or city names without an outline or backing.

There is also a big difference if you have borders on or off. With borders off it seems you need a higher luminance contrast than the relative hue/chroma contrast... adding it to my studies to do list... Legge had a study indicating that color contrast (hue chroma) do not add to the total contrast perception, and remains a separate contrast...

For readability, APCA looks at luminance only, as that is what is key for reading...

Thank you!

Andy
